### DISCUSSION TITLE
Paperless - add (optional) Tika & Gotenberg add-on ?

### DISCUSSION THREAD

**Discussion by Seqway** [community-scripts/ProxmoxVE#1112] [Q&A]
Hello,
i am running paperless script successfully. Everything is working fine.
To extend Paperless it is recommended to install Tika and Gotenberg while Gotenberg seems only to be available as docker image.
While paperless is a native install - is it possible to add the two add ons optional ?
i am facing a lot of problems to install it afterwards - up to now no success but I am also not a linux expert.
Would be nice to get some tips here.

**Comment by manuelkamp** [comment]
i also looked into this, for 2 years now. it seems noone has installed tika+gotenberg on bare metal, just docker (so did I). I also never got tika on bare metal working...maybe someone has already done this and can share his command history?

**Reply by Seqway** [reply]
Yes you are right!
May i ask you how you installed now Paperless wirh Tika and Gotenberg ?

I still hope that someone has a recommendation how to install it on a native Paperless install system without docker 😉

**Reply by jsapede** [reply]
i made de dokerstack for both, worked out of the box, only had to add the IP in the config of paperless

**Reply by Seqway** [reply]
What do you mean ? Did you make a separate container with docker and install there TIKA AND GOTENBERG ? If yes how =

**Reply by jsapede** [reply]
https://hub.docker.com/r/gotenberg/gotenberg/
https://hub.docker.com/r/apache/tika

worked out of the box 

you only have to specify the IP address of both containers into the paperless configs file

**Reply by Seqway** [reply]
> worked out of the box
> 
> you only have to specify the IP address of both containers into the paperless configs file

Question is how I install it ? I have Proxmox running with LXCs. Does it mean I need to install docker on a new LXC and then there install kia and Gutenberg ?
(by the way - I have docker)

**Reply by jsapede** [reply]
If you already have a docker simply add the containers, otherwise you install a docker lxc (docker inside lxc) with the scripts 
I have many docker stacks on différent lxc in my homelab for some apps (immich, frigate , romm, ...) or support fonctions (coturn, cupsd, ...)

**Reply by Seqway** [reply]
> https://hub.docker.com/r/gotenberg/gotenberg/ https://hub.docker.com/r/apache/tika
> 
> worked out of the box
> 
> you only have to specify the IP address of both containers into the paperless configs file

I installed docker separately and have both running but paperless does not get permission to connect it seems to be. I get an error !

`Error occurred while consuming document Testemail.eml: Could not parse content with tika server at http://192.168.40.126:9998: [Errno 111] Connection refused`

**Comment by andygrunwald** [comment]
I prefer both services as separate services, either VMs or Containers, and not part of the Paperless container.
Both services are helpful enough to be used by more services next to paperless.

## For others, here is a bit more information:

### Paperless Integration

See https://docs.paperless-ngx.com/configuration/#tika

### Apache Tika

> The Apache Tika™ toolkit detects and extracts metadata and text from over a thousand different file types (such as PPT, XLS, and PDF). All of these file types can be parsed through a single interface, making Tika useful for search engine indexing, content analysis, translation, and much more.

See https://tika.apache.org/

### Gotenberg

> Gotenberg provides a developer-friendly API to interact with powerful tools like Chromium and LibreOffice for converting numerous document formats (HTML, Markdown, Word, Excel, etc.) into PDF files, and more!

See https://gotenberg.dev/

## Similar discussions

See https://github.com/community-scripts/ProxmoxVE/discussions/499 and https://github.com/tteck/Proxmox/discussions/2286

**Reply by Seqway** [reply]
> I prefer both services as separate services, either VMs or Containers, and not part of the Paperless container.
> Both services are helpful enough to be used by more services next to paperless.

As above described I get an error and I have no clue what to do because I don't know docker at all :-(

`Error occurred while consuming document Testemail.eml: Could not parse content with tika server at http://192.168.40.126:9998: [Errno 111] Connection refused`

**Reply by jsapede** [reply]
let me have a look on mine in the afternoon

**Reply by Seqway** [reply]
> let me have a look on mine in the afternoon

Ok, sorry but I am really not familiar with docker.
Thanks in advance.
In parallel I try to setup a LXC and install Tika natively on that. Maybe this is better option for me later.

**Reply by jsapede** [reply]
deploying docker is oneline command

**Reply by Seqway** [reply]
> deploying docker is oneline command

yes I know. But I did it with the WebGUI and loaded from docker hub the image and then deposit it. As you can see I get the error and I can't reach the Tika via http.//x.y.z:9998 or 3000

**Comment by Citroene** [comment]
I installed paperlessngx lxc with the script in proxmox, in that same lxc i just installed docker with tika and Gotenberg containers. both then added in config file of paperlessngx. works like a charm!

My config file:
```
# Tika settings

PAPERLESS_TIKA_ENABLED=1
PAPERLESS_TIKA_ENDPOINT=http://x.x.x.x:9998
PAPERLESS_TIKA_GOTENBERG_ENDPOINT=http://x.x.x.x:3000
```

**Comment by DeepWoods** [comment]
I worked on this a bit last year and didn't get much response.   If it helps and someone wants to update it, here is the old discussion with the working code at the time.  https://github.com/tteck/Proxmox/discussions/2286#discussioncomment-8655426

**Comment by jsapede** [comment]
loiked at my install, gotenberg is in docker as per : https://hub.docker.com/r/gotenberg/gotenberg/

```
docker run --rm -p 3000:3000 gotenberg/gotenberg:8
```

and tika according to this : 

https://hub.docker.com/r/apache/tika

```
docker run -d -p 9998:9998 apache/tika:latest
```
Maybe here you should adapt the command of the dockerhub as if you let 127.0.0.1 in the port redirection you cannot use real IP of the TIKA, but only localhost

and written as is in paperless : 

![image](https://github.com/user-attachments/assets/f9aadacb-d85e-494b-8fab-5da87678f000)

with the IP of the LXC hosting the docker instance

**Comment by Vendo233** [comment]
Team how can I update Paperless

![image](https://github.com/user-attachments/assets/df9ab431-3e7a-4631-9dec-450dbf92d4c8)

I have tried command "update" it returns that it has been updated but it is not

![image](https://github.com/user-attachments/assets/1b27b3cf-ea1c-49e2-91b8-2958caf53670)

**Reply by MickLesk** [reply]
you use docker, not the paperless lxc. look @ paperless docs for docker support

**Reply by Vendo233** [reply]
> you use docker, not the paperless lxc. look @ paperless docs for docker support

thank you, I`m an idiot!

**Comment by andygrunwald** [comment]
Since today, Apache Tika is available as LXC Container, see https://github.com/community-scripts/ProxmoxVE/pull/2079
You can install as usual: https://community-scripts.github.io/ProxmoxVE/scripts?id=apache-tika

With the Docker LXC-Container at https://community-scripts.github.io/ProxmoxVE/scripts?id=docker, you can install the Gotenberg Docker container and run it.

I have both running and connected to Paperless. Docx documents can be indexed now.

**Comment by info-path** [comment]
The Paperless-ngx runs in an LXC container, Apache Tika runs in an LXC container, wouldn't it be nice if Gotenberg ran in a standalone LXC as well?

**Reply by andygrunwald** [reply]
I think this won't happen, see https://github.com/gotenberg/gotenberg/issues/1095

**Reply by info-path** [reply]
I see. I've run the Gotenberg in Docker and there's no problem with it but It's a shame it's not all in one place for maintenance reasons.

**Reply by supermarkert** [reply]
As an alternate, the Paperless devs have a Docker Compose stack of Paperless/Tika/Gotenberg on Github. So I ended up just creating a Docker LXC, then dropping in this compose file (with ENVs). Now it's all in one place.  
https://github.com/paperless-ngx/paperless-ngx/blob/dev/docker/compose/docker-compose.postgres-tika.yml
